webapi/Storage/Repository.cs

```diff
@@ -42,22 +42,21 @@ public Task DeleteAsync(T entity)
     }
 
     /// <inheritdoc/>
-    public Task<T> FindByIdAsync(string id)
+    public Task<T> FindByIdAsync(string id, string? partition = null)
     {
-        return this.StorageContext.ReadAsync(id);
+        return this.StorageContext.ReadAsync(id, partition ?? id);
     }
 
     /// <inheritdoc/>
-    public async Task<bool> TryFindByIdAsync(string id, Action<T?> entity)
+    public async Task<bool> TryFindByIdAsync(string id, string? partition = null, Action<T?>? callback = null)
```

Comment:
I think we need to pass the partition key, otherwise CosmosDb will have to scan all the partitions on every request